Custom charts: word smithing (#1131)
wandb/docs: content/guides/models/app/features/custom-charts/_index.md

- url: guides/app/features/custom-charts/:filename
---

Create custom charts in your W&B project. Log arbitrary tables of data and visualize them exactly how you want. Control details of fonts, colors, and tooltips with the power of [Vega](https://vega.github.io/vega/).

* **Code**: Try an example [Colab Colab notebook](https://tiny.cc/custom-charts).

> * **Video**: Watch a [walkthrough video](https://www.youtube.com/watch?v=3-N9OV6bkSM).

Reviewer: Nit: I don't think the bold labels are needed. Also the next one.

* **Example**: Quick Keras and Sklearn [demo notebook](https://colab.research.google.com/drive/1g-gNGokPWM2Qbc8p1Gofud0_5AoZdoSD?usp=sharing)

{{< img src="/images/app_ui/supported_charts.png" alt="Supported charts from vega.github.io/vega" max-width="90%" >}}

### How it works

1. **Log data**: From your script, log [config]({{< relref "/guides/models/track/config.md" >}}) and summary data.

> 2. **Customize the chart**: Pull in logged data with a [GraphQL](https://graphql.org) query. Visualize the results of your query with [Vega](https://vega.github.io/vega/), a powerful visualization grammar.

Reviewer: Same feedback.

3. **Log the chart**: Call your own preset from your script with `wandb.plot_table()`.

{{< img src="/images/app_ui/pr_roc.png" alt="" >}}

> If you do not see the data you are expecting, it might be because the column you are looking for is not logged in the runs you have selected. Save your chart and go back out to the runs table, and select the runs you would like to visualize with the **eye** icon.

Reviewer: ```suggestion
If you do not see the expected data, the column you are looking for might not be logged in the selected runs. Save your chart, go back out to the runs table, and verify selected runs using the **eye** icon.
```

## Log charts from a script

### Builtin presets

> W&B has a number of builtin chart presets that you can log directly from your script. These include line plots, scatter plots, bar charts, histograms, PR curves, and ROC curves.

Reviewer: ```suggestion
W&B includes builtin chart presets that you can log directly from your script: line plots, scatter plots, bar charts, histograms, PR curves, and ROC curves.
```

  )
  ```

>   You can use this to log curves on any two dimensions. Note that if you are plotting two lists of values against each other, the number of values in the lists must match exactly (for example, each point must have an x and a y).

Reviewer: ```suggestion
  A line plot logs curves on any two dimensions. If you plot two lists of values against each other, the number of values in the lists must match exactly (for example, each point must have an x and a y).
```
